scripts/seed-dev.sh

```
@@ -420,5 +420,63 @@ ensure_deal() { # ensure_deal <name> <stage-id> <amount-minor>
+  date -u -d "$1 days ago" +%Y-%m-%dT%H:%M:%SZ 2>/dev/null \
+    || date -u -v-"$1"d +%Y-%m-%dT%H:%M:%SZ
+}
+
+# Activities have no natural key, so idempotency is a subject probe against the
+# list — the same shape ensure_deal uses, for the same reason.
+ensure_conversation() { # ensure_conversation <subject> <kind> <direction> <days-ago> <person-id>…
+  local subject="$1" kind="$2" direction="$3" ago="$4"
+  shift 4
+  local links
+  links="$(printf '%s\n' "$@" | jq -R '{entity_type:"person",entity_id:.}' | jq -s '.')"
+  if [ -n "$(find_first "/activities?q=$(url_encode "$subject")" '.subject == $s' --arg s "$subject")" ]; then
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)